astroid/interpreter/objectmodel.py

```diff
@@ -142,7 +142,7 @@ def attr___new__(self) -> bases.BoundMethod:
         )
         # We set the parent as being the ClassDef of 'object' as that
         # triggers correct inference as a call to __new__ in bases.py
-        node.parent: nodes.ClassDef = AstroidManager().builtins_module["object"]
```

Comment:
Can't declare another object's attribute types

Reply:
Does `mypy` actually complain about this? If not it might be useful as additional documentation?

Reply:
`astroid/interpreter/objectmodel.py:145: error: Type cannot be declared in assignment to non-self attribute  [misc]`